inbox/crispin.py

```diff
@@ -877,6 +879,17 @@ def uids(self, uids: List[int]) -> List[RawMessage]:
                 # corrupt other UID data. Also we don't always get a message
                 # back at the first try.
                 for _ in range(3):
+                    # We already reject parsing messages bigger than MAX_MESSAGE_BODY_PARSE_LENGTH.
+                    # We might as well not download them at all, save bandwidth, processing time
+                    # and prevent OOMs due to fetching oversized emails.
+                    fetched_size: Dict[int, Dict[bytes, Any]] = self.conn.fetch(
+                        uid, ["RFC822.SIZE"]
+                    )
+                    body_size = int(fetched_size.get(uid, {}).get(b"RFC822.SIZE", 0))
+                    if body_size > MAX_MESSAGE_BODY_PARSE_LENGTH:
+                        log.warning("Skipping fetching of oversized message", uid=uid)
```

Comment:
Would it be possible to fetch message ID (and possibly other useful small-size bits of metadata) for this item and log it alongside the uid? It would seriously help debugging "email not synced to Close" tickets in the future.

Reply:
Good idea, Will follow up with another PR that fetches headers, parses them and adds some.